### PR TITLE
Make sure users are signed in when clicking on managed cards

### PR DESCRIPTION
**Pull request by iwiznia**
### Fixed Issues

$ https://github.com/Expensify/Expensify/issues/171698

### Tests
- Create a workspace via the global create button and give it a name
- Make sure you are signed out of expensify.com and staging.expensify.com 
- Go to the workspace settings and click on manage cards:
image
- Check that a new tab opens and it's in the https://www.expensify.com/domain_companycards (or https://staging.expensify.com/domain_companycards) and that you are logged in to the same account that in e.cash

### Tested On

- [X] Web
- [ ] Mobile Web
- [ ] Desktop
- [ ] iOS
- [ ] Android

### CODE REVIEW

File: src/CONST.js [hunk 1]
```diff
@@ -124,8 +124,12 @@ const CONST = {
     PRIVACY_URL: 'https://use.expensify.com/privacy',
     LICENSES_URL: 'https://use.expensify.com/licenses',
     PLAY_STORE_URL: 'https://play.google.com/store/apps/details?id=com.expensify.chat&hl=en',
-    ADD_SECONDARY_LOGIN_URL: 'settings?param={%22section%22:%22account%22}',
-    MANAGE_CARDS_URL: 'domain_companycards',
+    ADD_SECONDARY_LOGIN_URL: '/settings?param={%22section%22:%22account%22}',
+    VALIDATE_CODE_URL: (accountID, validateCode, exitTo = '') => {
```
Comment (cead22) [inline src/CONST.js:128]: This would be better in ROUTES, there are no functions in this file, and tons in ROUTES

File: src/ROUTES.js [hunk 2]
```diff
@@ -1,5 +1,11 @@
 import lodashGet from 'lodash/get';
+import {Linking} from 'react-native';
+import Onyx from 'react-native-onyx';
 import {addTrailingForwardSlash} from './libs/Url';
+import {GetAccountValidateCode} from './libs/API';
```
Comment (cead22) [inline src/ROUTES.js:5]: Nab but it's a bit weird for ROUTES to depend on API. Is there a better place for `openSignedInLink`?

File: src/ROUTES.js [hunk 3]
```diff
@@ -81,6 +95,19 @@ export default {
     WORKSPACE_INVITE: 'workspace/:policyID/invite',
     REQUEST_CALL: 'request-call',
 
+    /**
+     * This links to a page in e.com ensuring the user is logged in.
+     * It does so by getting a validate code and redirecting to the validate URL with exitTo set to the URL
+     * we want to visit
+     * @param {string} url relative URL to open in expensify.com
```
Comment (cead22) [inline src/ROUTES.js:102]: Does "relative" mean "with a leading slash", if so, can you add that explicitly?

**Review by cead22 — CHANGES_REQUESTED** [review]

File: src/ROUTES.js (hunk 2 above)
Comment (iwiznia) [reply src/ROUTES.js:5]: I don't know... tried putting it in URL but got the same shitty circular dependency.

Comment (iwiznia) [reply src/ROUTES.js:5]: Open to suggestions...

Comment (cead22) [reply src/ROUTES.js:5]: App.js

Comment (iwiznia) [reply src/ROUTES.js:5]: ok moved

**Review by cead22 — DISMISSED** [review]

**Comment by iwiznia** [comment]
Updated and retested (had not done it before the last push 😱 and it was broken)

**Review by cead22 — DISMISSED** [review]

**Comment by iwiznia** [comment]
Had conflicts, updated this. @cead22 please reapprove, @joelbettner please do your first review.

**Review by cead22 — APPROVED** [review]

**Review by joelbettner — APPROVED** [review]

**Pull request merged**